### PR TITLE
Add New Prompt Values to `auth0_prompt_custom_text` Resource

### PR DESCRIPTION
**Pull request by naari3**
### 🔧 Changes

- Added new prompt values to the `auth0_prompt_custom_text` resource.
- Updated the documentation to reflect the new prompt values.
- This change allows for more customization in the login flow by supporting additional prompts such as `captcha`, `custom-form`, `customized-consent`, `passkeys`, `phone-identifier-challenge`, and `phone-identifier-enrollment`.

### 📚 References

- https://auth0.com/docs/api/management/v2/prompts/put-custom-text-by-language

### 📝 Checklist

- [ ] All new/changed/fixed functionality is covered by tests (or N/A) N/A
- [x] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: internal/auth0/prompt/resource_custom_text.go [hunk 1]
```diff
@@ -17,10 +17,11 @@ import (
...
 	availablePrompts = []string{
-		"common", "consent", "device-flow", "email-otp-challenge", "email-verification", "invitation", "login",
-		"login-id", "login-password", "login-passwordless", "login-email-verification", "logout", "mfa", "mfa-email",
-		"mfa-otp", "mfa-phone", "mfa-push", "mfa-recovery-code", "mfa-sms", "mfa-voice", "mfa-webauthn",
-		"organizations", "reset-password", "signup", "signup-id", "signup-password", "status",
+		"capcha", "common", "consent", "custom-form", "customized-consent", "device-flow", "email-otp-challenge",
+		"email-verification", "invitation", "login", "login-id", "login-password", "login-passwordless",
+		"login-email-verification", "logout", "mfa", "mfa-email", "mfa-otp", "mfa-phone", "mfa-push", "mfa-recovery-code",
+		"mfa-sms", "mfa-voice", "mfa-webauthn", "organizations", "passkeys", "phone-identifier-challenge",
+		"phone-identifier-enrollment", "reset-password", "signup", "signup-id", "signup-password", "status",
```
Comment (developerkunal) [inline internal/auth0/prompt/resource_custom_text.go:24]:
```suggestion
		"captcha", "common", "consent", "custom-form", "customized-consent", "device-flow", "email-otp-challenge",
		"email-verification", "invitation", "login", "login-email-verification", "login-id", "login-password",
		"login-passwordless", "logout", "mfa", "mfa-email", "mfa-otp", "mfa-phone", "mfa-push", "mfa-recovery-code",
		"mfa-sms", "mfa-voice", "mfa-webauthn", "organizations", "passkeys", "phone-identifier-challenge",
		"phone-identifier-enrollment", "reset-password", "signup", "signup-id", "signup-password", "status"
```

File: docs/resources/prompt_custom_text.md [hunk 2]
```diff
@@ -49,7 +49,7 @@ resource "auth0_prompt_custom_text" "example" {
...
-- `prompt` (String) The term `prompt` is used to refer to a specific step in the login flow. Options include: `common`, `consent`, `device-flow`, `email-otp-challenge`, `email-verification`, `invitation`, `login`, `login-id`, `login-password`, `login-passwordless`, `login-email-verification`, `logout`, `mfa`, `mfa-email`, `mfa-otp`, `mfa-phone`, `mfa-push`, `mfa-recovery-code`, `mfa-sms`, `mfa-voice`, `mfa-webauthn`, `organizations`, `reset-password`, `signup`, `signup-id`, `signup-password`, `status`.
+- `prompt` (String) The term `prompt` is used to refer to a specific step in the login flow. Options include: `capcha`, `common`, `consent`, `custom-form`, `customized-consent`, `device-flow`, `email-otp-challenge`, `email-verification`, `invitation`, `login`, `login-id`, `login-password`, `login-passwordless`, `login-email-verification`, `logout`, `mfa`, `mfa-email`, `mfa-otp`, `mfa-phone`, `mfa-push`, `mfa-recovery-code`, `mfa-sms`, `mfa-voice`, `mfa-webauthn`, `organizations`, `passkeys`, `phone-identifier-challenge`, `phone-identifier-enrollment`, `reset-password`, `signup`, `signup-id`, `signup-password`, `status`.
```
Comment (developerkunal) [inline docs/resources/prompt_custom_text.md:52]: Please don't make manual changes. Ensure you are using `make docs` to generate and `make check-docs` to verify the documentation.

**Review by developerkunal — CHANGES_REQUESTED** [review]
Hi @naari3,

I have added comments and changes. Please follow them and make the necessary fixes.

File: docs/resources/prompt_custom_text.md (hunk 2 above)
Comment (naari3) [reply docs/resources/prompt_custom_text.md:52]: ah sorry, I ran `make docs` but not `check-docs`. Is this diff specific to manually creating docs?

File: internal/auth0/prompt/resource_custom_text.go (hunk 1 above)
Comment (naari3) [reply internal/auth0/prompt/resource_custom_text.go]: It seems to be invalid without a trailing comma, so I'll added it
```
terraform-provider-auth0 on  add-new-prompt-values-to-auth0_prompt_custom_text via 🐹 v1.22.1 on ☁️   
❯ make docs
 ::  Generating docs                                                              
rendering website for provider "terraform-provider-auth0" (as "terraform-provider-auth0")
copying any existing content to tmp dir
exporting schema from Terraform
compiling provider "auth0"
2024/07/12 18:45:21 error executing "/usr/local/Cellar/go/1.22.1/libexec/bin/go", [go build -o /var/folders/2t/kxlpb5797yx1zccty0bg42vwb_bpfs/T/tfws2095941905/plugins/registry.terraform.io/hashicorp/auth0/0.0.1/darwin_amd64/terraform-provider-auth0]
2024/07/12 18:45:21 # github.com/auth0/terraform-provider-auth0/internal/auth0/prompt
internal/auth0/prompt/resource_custom_text.go:24:102: syntax error: unexpected newline in composite literal; possibly missing comma or }
Error executing command: unable to generate website: error exporting provider schema from Terraform: unable to execute go build command: error executing "/usr/local/Cellar/go/1.22.1/libexec/bin/go": exit status 1

exit status 1
main.go:14: running "go": exit status 1
make: *** [docs] Error 1
```

Comment (developerkunal) [reply internal/auth0/prompt/resource_custom_text.go]: Sure!

File: docs/resources/prompt_custom_text.md (hunk 2 above)
Comment (developerkunal) [reply docs/resources/prompt_custom_text.md]: No, But we recommend if we do changes like this we run `make docs`, and we also check generated docs using `make check-docs`.

**Comment by naari3** [comment]
@developerkunal Addressed the comments and pushed updates. Please review again. Thanks!

**Review by developerkunal — APPROVED** [review]
Hi @naari3,

Thanks for the contribution! Your changes look good to me.

**Pull request merged**